meshnets/modules/processor.py

```diff
@@ -1,22 +1,75 @@
 """"Define the Processor and ProcesserLayer classes."""
 
 import torch
-from torch.nn import Identity
+from torch.nn import Sequential
 from torch_geometric.data import Batch
 from torch_geometric.nn.conv import MessagePassing
+import torch_scatter
+
+from meshnets.modules.mlp import MLP
 
 
 class MGNProcessorLayer(MessagePassing):
-    """Single MGN Message Passing layer for a Graph object."""
+    """Single MGN Message Passing layer for graphs."""
+
+    def __init__(self, latent_size, num_mlp_layers):
+        """Initialize a processor layer given the latent size
+        and number of layers in the MLPs.
 
-    def __init__(self, latent_size):
+        Note that the latent node features and latent edge features
+        have the same size. Thus, the input of the edge processor is
+        three times the hidden dimension (one self embedding and two
+        adjacent node embeddings). The input of the node processor is
+        two times the hidden dimension (one self embedding and one sum
+        of connected edges embedings)
```

Comment:
It doesn't necessarily have to be the "sum of connected edges embeddings", right? It could be some other aggregation resulting in a scalar. Btw, there is a typo in "embedings".

Reply:
In this case it is a sum, but I changed it to be more general. Also fixed the typo

Reply:
In this case it is a sum, but I changed it to be more general. Also fixed the typo